pkg/apis/nsx.vmware.com/v1alpha1/ipaddressallocation_types.go

```diff
@@ -0,0 +1,58 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+
+// IPAddressAllocation is the Schema for the ipallocation API.
+// +kubebuilder:printcolumn:name="IPAddressBlockVisibility",type=string,JSONPath=`.spec.ip_address_block_visibility`,description="IPAddressBlockVisibility of IPAddressAllocation"
+// +kubebuilder:printcolumn:name="CIDR",type=string,JSONPath=`.status..cidr`,description="CIDRs for the IPAddressAllocation"
+type IPAddressAllocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   IPAddressAllocationSpec   `json:"spec"`
+	Status IPAddressAllocationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// IPAddressAllocationList contains a list of IPAddressAllocation.
+type IPAddressAllocationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IPAddressAllocation `json:"items"`
+}
+
+// IPAddressAllocationSpec defines the desired state of IPAddressAllocation.
+type IPAddressAllocationSpec struct {
+	// IPAddressBlockVisibility defines the visibility of IP address block, External or Private.
+	// +kubebuilder:validation:Enum=External;Private
+	// +optional
+	IPAddressBlockVisibility string `json:"ip_address_block_visibility:,omitempty"`
+	IPBlock                  string `json:"ip_block,omitempty"`
+	// AllocationSize defines allocation size for auto allocate IP based on
+	// +optional
+	AllocationSize int `json:"allocation_size,omitempty"`
+	// AllocationSize defines CIDR or range of IPs that needs to be reserved
```

Comment:
AllocationSize -> AllocationIPs

Do we need this field? If not we can remove it now.

Reply:
+1, let's remove the properties which we do not need in current feature

Reply:
ok, removed.